apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/configuration/AuthConfiguration.java

```diff
@@ -111,27 +117,35 @@ public LogoutHandler logoutHandler() {
     }
 
     @Bean
-    public static JdbcUserDetailsManager jdbcUserDetailsManager(PasswordEncoder passwordEncoder,
-        AuthenticationManagerBuilder auth, DataSource datasource) throws Exception {
+    public static JdbcUserDetailsManager jdbcUserDetailsManager(
+            PasswordEncoder passwordEncoder,
+            AuthenticationManagerBuilder auth,
+            DataSource datasource,
+            EntityManagerFactory entityManagerFactory) throws Exception {
+      char openQuote = '`';
+      char closeQuote = '`';
+      try {
+        SessionFactoryImplementor sessionFactory = entityManagerFactory.unwrap(
+                SessionFactoryImplementor.class);
+        Dialect dialect = sessionFactory.getJdbcServices().getDialect();
+        openQuote = dialect.openQuote();
+        closeQuote = dialect.closeQuote();
+      } catch (Throwable ex) {
+        //ignore
+      }
       JdbcUserDetailsManager jdbcUserDetailsManager = auth.jdbcAuthentication()
-          .passwordEncoder(passwordEncoder).dataSource(datasource)
-          .usersByUsernameQuery("select Username,Password,Enabled from `Users` where Username = ?")
-          .authoritiesByUsernameQuery(
-              "select Username,Authority from `Authorities` where Username = ?")
-          .getUserDetailsService();
-
-      jdbcUserDetailsManager.setUserExistsSql("select Username from `Users` where Username = ?");
-      jdbcUserDetailsManager
-          .setCreateUserSql("insert into `Users` (Username, Password, Enabled) values (?,?,?)");
-      jdbcUserDetailsManager
-          .setUpdateUserSql("update `Users` set Password = ?, Enabled = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
-      jdbcUserDetailsManager.setDeleteUserSql("delete from `Users` where id = (select u.id from (select id from `Users` where Username = ?) as u)");
-      jdbcUserDetailsManager
-          .setCreateAuthoritySql("insert into `Authorities` (Username, Authority) values (?,?)");
-      jdbcUserDetailsManager
-          .setDeleteUserAuthoritiesSql("delete from `Authorities` where id in (select a.id from (select id from `Authorities` where Username = ?) as a)");
-      jdbcUserDetailsManager
-          .setChangePasswordSql("update `Users` set Password = ? where id = (select u.id from (select id from `Users` where Username = ?) as u)");
+              .passwordEncoder(passwordEncoder).dataSource(datasource)
+              .usersByUsernameQuery(usersQuerySql(openQuote, closeQuote))
+              .authoritiesByUsernameQuery(authoritiesQuerySql(openQuote, closeQuote))
+              .getUserDetailsService();
+
+      jdbcUserDetailsManager.setUserExistsSql(usersExistsSql(openQuote, closeQuote));
```

Comment:
How about we simply inline the statement here as the method is not used elsewhere.